external/llvm-project/mlir/lib/Conversion/GPUToROCDL/LowerGpuOpsToROCDLOps.cpp

```diff
@@ -295,18 +295,23 @@ struct MubufLoadOpLowering : ConvertToLLVMPattern {
         loc, LLVMI1Type, rewriter.getIntegerAttr(I1Type, 0));
 
     if (srcElementType.getIntOrFloatBitWidth() != 32) {
-      // for f16 and i16 (bf16) types, use f32 buffer_load and bitcast the
-      // result.
+      // For non-f32 types, use f32 buffer_load and bitcast the result.
       assert(resultType.isa<VectorType>());
-      // deduce the interim type for f16 / i16 (bf16).
       VectorType vectorResultType = resultType.template cast<VectorType>();
       auto vectorShape = vectorResultType.getShape();
 
+      // For int8, each 32 bits will fit 4;
+      // For f16 and i16, each 32 bits will fit 2
+      int64_t numFit32 = 32 / vectorResultType.getElementTypeBitWidth();
+      int64_t lastDimShape = vectorShape.back() / numFit32;
+
       SmallVector<int64_t, 1> interimShape;
       for (unsigned iter = 0; iter < vectorShape.size() - 1; ++iter)
         interimShape.push_back(vectorShape[iter]);
-      interimShape.push_back(vectorShape[vectorShape.size() - 1] >> 1);
-      bool useScalarF32 = (vectorShape.size() == 1) && (vectorShape[0] == 2);
+      interimShape.push_back(lastDimShape);
+      bool useScalarF32 =
```

Comment:
Minor nit: since there's bitcasts and such involved, could this be an `i32` load instead of an `f32` load? It'll make reading the resulting IR clearer.

Reply:
Yeah, I think I did this implicitly.